### ISSUE TITLE
People profiles 

### ISSUE THREAD

**Issue by asimonw** [emfoundation/ce100-app#1147]
As a user of cello (primary or secondary), I want to have a personal profile page, so that people can easily find my details and see the challenges that I have shared.

**Acceptance criteria**
+ [x] the header contains all the information which currently displayed for people in the People list
+ [x] If no phone number is provided, don't show the phone icon
+ [x] If there's no job title, don't display the "at"
+ [x] If there are no challenges, display same message as currently on org profile
+ [x] An edit icon in the top right corner points to the settings page of the user
+ [x] Button to add a challenge

+ [x] Create profile page **only** for primary and secondary users (users who have a link to an org)
+ [x] under the header you'll see a list of challenges shared by the person, every item a link to the individual challenge page
+ [x] The creator and the admin of the challenge can see their own archived challenges from the 'my archived challenges' link
+ [x] secondary users can create challenges
+ [x] primary users can create challenges

**Comment by fannyBkane** [comment]
[Link to invision - desktop](https://projects.invisionapp.com/d/main#/console/12767821/267455656/preview) ![screen shot 2017-12-05 at 09 34 55](https://user-images.githubusercontent.com/20577339/33599915-97c286f8-d99f-11e7-91b2-f2e522cfda3e.png)

**Comment by Cleop** [comment]
@fannyBkane @asimonw 
> the header contains all the information which currently displayed for people in the People list

- does this mean the user type ie. primary, secondary that are not currently in the invision mock up?

**Comment by fannyBkane** [comment]
@Cleop @asimonw thanks for pointing that out :)
I've updated the link and here is the screenshot:

There is a tiny difference in the order in which the info is displayed compared to the people list, it makes more sense to display 'user type' after the name. It could but _doesn't have to_ be updated on the people list. Thank you!
![screen shot 2017-12-06 at 09 24 19](https://user-images.githubusercontent.com/20577339/33654341-aee339ea-da67-11e7-8c27-b8ea83802558.png)

**Comment by asimonw** [comment]
Thanks @fannyBkane, I think it looks good there on the profile page.
I can't be absolutely sure, but it might be easier to keep it where it is on the list views because then the other content doesn't need to shift between admin and other users.

**Comment by Cleop** [comment]
@asimonw @fannyBkane - what would you expect users to see if they typed in the url that they might expect would take them to the profile page of an admin? 

At the moment it would be a 'Sorry that page is not available' message. I just wanted to check that that was what you had in mind.

I wonder if users may not know why EMF/admin members names are not clickable in lists and why they don't have a profile. Also, based on the new people list data designs, would this mean that EMF staff's phone numbers would only be visible on mobile but no other context?

**Comment by Cleop** [comment]
**Notes for Cleo and @SimonLab only**--- ignore
- Create profile page **only** for primary and secondary users (users who have a link to an org) 
- at the moment the /profile/{id} will take you to any url after '/profile' because id is not defined. How can we get around this? regex?

**Comment by asimonw** [comment]
@Cleop Maybe something important to realise is that admin and other internal role users are not visible to other users in the people list and they should not be attached to orgs, so should also not show up there (I've noticed that this has already happened, but will be avoided going forward). So "front-end" users should never see those "back-end" users. They are not meant to participate on the platform.

One exception right now is that admins etc can also comment on challenges. I'm not too happy about this inconsistency, so we're thinking about a solution for this. (One option might be to see front-end and back-end roles as orthogonal, in the sense that one user can have both a front-end and a back-end role). For now, comments made by back-end users (admin, content owner, view only) should not link to any profile, which will indeed feel a bit odd. This is already mentioned in #1162 

So long story short, displaying a page not found is good 😄

**Comment by asimonw** [comment]
@Cleop That reminded me to do a quick check and I notice now that admins are indeed hidden, but content owners and view only users are not! This was not the intention, but must have slipped our radar. I've created issue #1164 to deal with this.

@iteles Ideally this issue is resolved before we ship user profiles, so I've added it to Sprint 12. It shouldn't be a huge one, but let me know what the impact is.

**Comment by fannyBkane** [comment]
> Would you either like me to make the masthead taller in this instance sacrificing the consistency OR would you like me to make some of the components smaller (maybe the picture) so the masthead can stay the consistent size?

@Cleop sorry I've missed that comment! Your suggestion (to make the picture smaller) makes sense. One risk is that the whole thing loses balance... But we won't know until we see!

**Comment by asimonw** [comment]
@Cleop Looks pretty good! I did spot a few extra things which we forgot in the criteria
+ The challenge list should have an edit button for every challenge (like on the org profile)
+ After saving the settings page and canceling an "add challenge" action, the user should be redirected to their profile page, no longer to the org profile page
+ This was going to be a new issue, but maybe it's not really necessary: improve spacing between user name and job title in the header

**Comment by fannyBkane** [comment]
@Cleop It all looks good! I suppose we ave the spacing issue on our radar as it is referred to in #1201.
A tiny thing that was not part of the acceptance criteria, but the + Challenge button looks awkward when aligned with the header, is it possible to add some margin right to align it with the challenge titles instead? Happy to create a new issue for this @asimonw  ?

**Comment by asimonw** [comment]
@fannyBkane Yes please create a separate issue for that, so it can be prioritised separately. I'll close this one.
